gpiozero/output_devices.py

```diff
@@ -562,8 +562,12 @@ def __init__(
             raise GPIOPinMissing('red, green, and blue pins must be provided')
         LEDClass = PWMLED if pwm else LED
         super(RGBLED, self).__init__()
-        self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
-        self.value = initial_value
+        try:
+            self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
+            self.value = initial_value
+        except:
+            self.close()
+            raise
```

Comment:
Basically it's needed whenever you construct some subordinate Device, then fail to close those subordinates if your own constructor subsequently fails. LEDCollection was definitely an example where that was needed. ButtonBoard needs it too. As it happens SnowPi and TrafficLights don't need it because LEDBoard descends from LEDCollection and that's handling clean-up of any failed LED constructions. FishDish needs it because it constructs its own Buzzer and Button, same story for TrafficHat (note: TrafficLights is *not* the issue here, just the Buzzer and Button). Robot doesn't need it, but the subordinate Motor class does, and so does Servo and _EnergenieMaster. 

I'll get all these fixed up.